frontend/front/src/pages/Public/Pages/SurveyPage.js

```diff
@@ -76,14 +79,33 @@ export const SurveyPage = () => {
     [addSurveyVisit, getReCaptchaToken]
   );
 
+  const isCanonicalized = validationStatus === "canonicalized";
+
   const step = useMemo(() => {
-    if (!createHomeData && !isSurveyVisitSuccess) {
+    if (!isCanonicalized && !isSurveyVisitSuccess) {
       return STEP_ADDRESS;
     } else if (isSurveyVisitSuccess) {
       return STEP_THANKS;
+    } else if (!!createHomeData && isCanonicalized) {
+      return STEP_SURVEY;
     }
-    return STEP_SURVEY;
-  }, [createHomeData, isSurveyVisitSuccess]);
+  }, [createHomeData, isSurveyVisitSuccess, isCanonicalized]);
+
+  const showValidationLoader = useMemo(() => {
+    if (!createHomeData?.id) return false;
+    if (step !== STEP_ADDRESS) return false;
+    if (
+      validationStatus === "uncanonicalized" ||
+      validationStatus === undefined
+    )
+      return true;
+    return false; // For completed validations (unrecognized or canonicalized).
+  }, [createHomeData, validationStatus, step]);
```

Comment:
Loader visible after the home is created but canonicalization status hasn't yet been determined.